sources/packages/api/src/services/institution/institution.service.ts

```diff
@@ -447,27 +444,37 @@ export class InstitutionService extends RecordDataModelService<Institution> {
    * @param institutionUserId institution user to be updated.
    * @param permissions complete list of the user permissions that will entirely
    * replace the existing ones.
+   * @param auditUserId user that should be considered the one that is causing the changes.
+   * @param ensureHasAdmin if true, will ensure that at least one admin is present, otherwise
+   * will allow that all institution admins are removed or disabled.
    */
   async updateInstitutionUser(
     institutionId: number,
     institutionUserId: number,
     permissions: UserPermissionModel[],
+    auditUserId: number,
+    ensureHasAdmin = true,
   ): Promise<void> {
-    const validateAtLeastOneAdmin = this.validateAtLeastOneAdmin(
-      institutionId,
-      institutionUserId,
-      permissions,
-    );
-    const validateUniqueSigningAuthority = this.validateUniqueSigningAuthority(
-      institutionId,
-      permissions,
-      institutionUserId,
+    const validations: Promise<void>[] = [];
```

Comment:
for my knowledge, what is the benefit of `:Promise<void>[] = []` with `:[] = []`

Reply:
If I understood the question, `:[] = []` will be an array of `any` and `Promise<void>[]` will be a typed array of `:Promise<void>`.

Reply:
yup, you are correct. ignore it. 😉